idea/src/org/jetbrains/kotlin/idea/intentions/branchedTransformations/branchedTransformationUtils.kt

```diff
@@ -65,28 +59,32 @@ fun KtWhenExpression.getSubjectToIntroduce(checkConstants: Boolean = true): KtEx
     return lastCandidate
 }
 
-private fun KtExpression?.getWhenConditionSubjectCandidate(checkConstants: Boolean): KtExpression? = when (this) {
-    is KtIsExpression -> leftHandSide
-
-    is KtBinaryExpression -> {
-        val lhs = left
-        val rhs = right
-        when (operationToken) {
-            KtTokens.IN_KEYWORD, KtTokens.NOT_IN -> lhs
-            KtTokens.EQEQ ->
-                lhs?.takeIf { it.hasCandidateNameReferenceExpression(checkConstants) }
-                    ?: rhs?.takeIf { it.hasCandidateNameReferenceExpression(checkConstants) }
-            KtTokens.OROR -> {
-                val leftCandidate = lhs.getWhenConditionSubjectCandidate(checkConstants)
-                val rightCandidate = rhs.getWhenConditionSubjectCandidate(checkConstants)
-                if (leftCandidate.matches(rightCandidate)) leftCandidate else null
+fun KtExpression?.getWhenConditionSubjectCandidate(checkConstants: Boolean): KtExpression? {
+    fun KtExpression?.getCandidate(): KtExpression? = when (this) {
+        is KtIsExpression -> leftHandSide
+        is KtBinaryExpression -> {
+            val lhs = left
+            val rhs = right
+            when (operationToken) {
+                KtTokens.IN_KEYWORD, KtTokens.NOT_IN -> lhs
+                KtTokens.EQEQ ->
+                    lhs?.takeIf { it.hasCandidateNameReferenceExpression(checkConstants) }
+                        ?: rhs?.takeIf { it.hasCandidateNameReferenceExpression(checkConstants) }
+                KtTokens.OROR -> {
+                    val leftCandidate = lhs.getCandidate()
+                    val rightCandidate = rhs.getCandidate()
+                    if (leftCandidate.matches(rightCandidate)) leftCandidate else null
+                }
+                else -> null
             }
-            else -> null
         }
-
+        else -> null
+    }
+    return getCandidate()?.takeIf {
+        it is KtNameReferenceExpression
```

Comment:
This doesn't make much sense. If all expressions excluding `KtThisExpression` match, the `it is KtNameReferenceExpression` check is useless. It looks like a mistake in De Morgan's law application.